rfd/0153-resource-guidelines.md

```diff
@@ -22,6 +22,68 @@ operator that have to accommodate for every subtle API and resource difference t
 
 ## Details
 
+### Project structure
+
+All protos are defined in the api module under the proto directory. When adding a new resource and gRPC API a new folder
+that matches the desired package name of the proto, generally a friendly name of the resource, should be created. Inside
+that a directory for each version of the API should exist. The actual RPC service should exist in its own file
+`foo_service.proto` which has the service defined first and all request/response messages defined after. This allows to
+quickly discover the API without having to scroll through a bunch of boilerplate and standard messages. Important types,
+like the resource definition, or any supporting types should exist in their own file. This makes discovering them easier
+and reduces the amount of things required to be imported.
+
+An example of the file layout for the resource used as an example in this RFD is included below.
+
+```bash
+api/proto
+├── README.md
+├── buf-legacy.yaml
+├── buf.lock
+├── buf.yaml
+└── teleport
+    ├── foo
+    │   └── v1
+    │       ├── foo.proto
+    │       └── foo_service.proto
+    ├── legacy
+    │   ├── client
+    │   │   └── proto
+    │   │     ├── authservice.proto
+    │   │     ├── certs.proto
+    │   │     ├── event.proto
+    │   │     ├── joinservice.proto
+    │   │     └── proxyservice.proto
+    │   └── types
+    │         ├── device.proto
+    │         ├── events
+    │         │    ├── athena.proto
+    │         │    └── events.proto
+    │         ├── types.proto
+    │         ├── webauthn
+    │         │    └── webauthn.proto
+    │         └── wrappers
+    │              └── wrappers.proto
+```
+
+The legacy directory contains resources and API definitions that were defined prior to our shift to user smaller,
+localized services per resource. Adding new resources and APIs to the giant monolithic `proto.AuthService` should be
+avoided if possible.
+
+### New resource checklist
+
+Use the non-exhaustive list below as a guide when adding a new resource. Further sections in the RFD dive into more
+detail about what is needed to complete a particular item.
+
+- [ ] Create proto spec for resource and RPC service
+- [ ] Create backend service
+- [ ] Add resource support to api client
```

Comment:
Could we avoid that and, instead, add the new gRPC client itself to the API client? It makes the API client smaller and does away with a bunch of boilerplate.

Ie: https://github.com/gravitational/teleport/blob/b65e7f3b7e864a9e0737922948b46ef9e9d8675a/api/client/client.go#L730-L732

Reply:
Gotcha. I think it's worth spelling out, otherwise we might see method duplication there (as was the norm).